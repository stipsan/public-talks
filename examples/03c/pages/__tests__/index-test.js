import renderer from 'react-test-renderer'
import App from '../index'

jest.mock('shared/Footer', () => {
  // Instead of mocking we're just overriding the default value of a prop
  const { Footer } = require.requireActual('shared/Footer')

  return { Footer: props => <Footer {...props} year={2049} /> }
})

test('App renders correctly', () => {
  const component = renderer.create(<App />)
  expect(component.toJSON()).toMatchSnapshot()
})
