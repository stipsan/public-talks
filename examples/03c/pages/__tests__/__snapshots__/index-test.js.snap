// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <p>
    Hello 
    ReactJS Oslo!
    !
  </p>
  <footer>
    Â© 
    2049
     
    ACME Inc.
  </footer>
</div>
`;
